index.js

```diff
@@ -42,6 +42,10 @@ function expressPlugin (fastify, options, next) {
     req.raw.ips = req.ips
     req.raw.log = req.log
     reply.raw.log = req.log
+    // added in Fastify@3.5
+    if (req.protocol) {
```

Comment:
I'm not sure this guard is needed?

Reply:
protocol is either `http` or `https`, it's always defined.

Reply:
yes, but only after https://github.com/fastify/fastify/pull/2582, right? So if the user is on an older version of Fastify it won't be set. I don't know if it matters that we then set `req.raw.protocol` to `undefined`?

Reply:
oh, and if `trustProxy` is active it's not necessarily `http` or `http` since Fastify respects the incoming `X-Forwarded-Host` header. Not sure if the TS type should change or not

Reply:
Note that this looks pretty slow (unfortunately): https://github.com/fastify/fastify/blob/a7948798fb88a2054c65f123996f53714ef2b1da/lib/request.js#L83-L92, so maybe cache it? Or you can define it as a getter on `req` as well.

Reply:
made it into a getter